contrib/globalsign/mgo/mgo_test.go

```diff
@@ -0,0 +1,355 @@
+	defer session.Close()
+
+	assert.NotNil(session)
+	assert.Nil(err)
+
+	db := session.DB("my_db")
+	collection := db.C("MyCollection")
+
+	command(collection)
+
+	parentSpan.Finish()
+
+	spans := mt.FinishedSpans()
+	return spans
+}
+
+func TestCollection_Insert(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(2, len(spans))
+	assert.Equal("mongodb.query", spans[0].OperationName())
+}
+
+func TestCollection_Update(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		collection.Update(entity, entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(3, len(spans))
+	assert.Equal("mongodb.query", spans[1].OperationName())
+}
+
+func TestCollection_UpdateId(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		var r bson.D
+		collection.Find(entity).Iter().Next(&r)
+		collection.UpdateId(r.Map()["_id"], entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(5, len(spans))
+	assert.Equal("mongodb.query", spans[3].OperationName())
+}
+
+func TestCollection_Upsert(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		collection.Upsert(entity, entity)
+		var r bson.D
+		collection.Find(entity).Iter().Next(&r)
+		collection.UpsertId(r.Map()["_id"], entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(6, len(spans))
+	assert.Equal("mongodb.query", spans[1].OperationName())
+	assert.Equal("mongodb.query", spans[4].OperationName())
+}
+
+func TestCollection_UpdateAll(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		collection.UpdateAll(entity, entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(3, len(spans))
+	assert.Equal("mongodb.query", spans[1].OperationName())
+}
+
+func TestCollection_FindId(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
```

Comment:
There is a big amount of duplication in these tests. We should compact the tests as much as possible (by abstracting away all the logic) in such a way that it is clearly visible what the list of input and output is without scrolling through the file.

Make generic functions from duplicated code and try to create a minimal amount of tests using [sub-tests](https://blog.golang.org/subtests). Try to accomplish a table, if possible.